component/azstorage/azstorage.go

```diff
@@ -352,10 +352,15 @@ func (az *AzStorage) ReadInBuffer(options internal.ReadInBufferOptions) (length
 }
 
 func (az *AzStorage) WriteFile(options internal.WriteFileOptions) (int, error) {
-	err := az.storage.WriteFromBuffer(options.Handle.Path, nil, options.Data)
+	err := az.storage.Write(options.Handle.Path, options.Offset, int64(len(options.Data)), options.Data, options.FileOffsets, options.ModBlockList)
```

Comment:
Here we are updating the actual content on container with every write and not wait for flush/close. We need to document this that even after doing write if you just ignore close/flush call from application data will be persisted. This does not happen with regular file-cache based flow as we upload only on flush/close.

Reply:
yes the azstorage component and this `write` code is stateless. In the stream layer we can delay flushes.

Reply:
Yes I understood the flow but my suggestion is to document this part clearly in the README under some streaming section that 1. it will update file on every write 2. it will not be very performant if new file is getting created or file is written serially

Reply:
Will mention in the README.